pkg/apis/istio/v1alpha1/istioinstaller_types.proto

```diff
@@ -0,0 +1,441 @@
+    CommonComponentSpec common = 1;
+}
+
+// Configuration options for node agent component.
+message NodeAgentComponentSpec {
+    CommonComponentSpec common = 1;
+}
+
+// Configuration options for node agent component.
+message GalleyComponentSpec {
+    CommonComponentSpec common = 1;
+}
+
+// Configuration common to all components.
+message CommonComponentSpec {
+    // Selects whether this component is installed.
+    google.protobuf.BoolValue enabled = 1;
+    // Namespace that all resources for this component are installed into.
+    string namespace = 2;
+    // LogLevel sets the component log level. Mappings to underlying log levels are not necessarily 1:1 and depend on
+    // component support of these levels e.g. DEBUG or INFO may result in the same log verbosity.
+    enum LogLevel {
+        NONE = 0;
+        ERROR = 1;
+        WARN = 2;
+        DEBUG = 3;
+        INFO = 4;
+    }
+    LogLevel debug = 3;
+
+    // Kubernetes resource spec.
+    KubernetesResourcesSpec k8s = 80;
+
+    // Kustomize style overrides for default values.yaml.
+    // ValuesOverrides map[string]interface{} `protobuf:"bytes,6,opt,name=data" json:"valuesOverrides,omitempty"`
+}
+
+// KubernetesResourcesConfig is a common set of k8s resource configs for components.
+message KubernetesResourcesSpec {
+    // k8s resources settings.
+    // https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#resource-requests-and-limits-of-pod-and-container
+    Resources resources = 1;
+    // k8s readinessProbe settings.
+    // https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/
+    k8s.io.api.core.v1.Probe readiness_probe = 2;
+    // k8s Deployment replicas setting.
+    // https://kubernetes.io/docs/concepts/workloads/controllers/deployment/
+    uint32 replica_count = 3;
+    // k8s HorizontalPodAutoscaler settings.
+    // https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/
+    k8s.io.api.autoscaling.v1.HorizontalPodAutoscalerSpec hpa_spec = 4;
+    // k8s PodDisruptionBudget settings.
+    // https://kubernetes.io/docs/concepts/workloads/pods/disruptions/#how-disruption-budgets-work
+    PodDisruptionBudgetSpec pod_disruption_budget = 5;
+    // k8s pod annotations.
+    // https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/
+    map<string, string> pod_annotations = 6;
+    // k8s nodeSelector.
+    // https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector
+    map<string, string> node_selector = 7;
+    // k8s ImagePullPolicy setting for all containers.
+    // https://kubernetes.io/docs/concepts/containers/images/#updating-images
+
+    // k8s node affinity.
+    // https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
+    message NodeAffinity {
+        uint32 arch_amd64 = 1;
+        uint32 arch_s390x = 2;
+        uint32 arch_ppc64le = 3;
+    }
```

Comment:
we also have rolling update strategy and pod affinity.

Reply:
also, cpu/mem limit and requested values

Reply:
requests/limits are already under Resources above. 
Added k8s Affinity which takes care of both pod and node [anti-]affinity.